docs/using-modules.rst

```diff
@@ -0,0 +1,192 @@
+.. _modules:
+
+Modules
+#######
+
+A module is a set of function definitions and variable declarations which enables code reuse. Vyper favors code reuse through composition, rather than inheritance. A module encapsulates everything needed for code reuse, from type and function declarations to state.
+
+Declaring and using modules
+===========================
+
+The simplest way to define a module is to write a contract. In Vyper, any contract is a valid module! For example, the following contract is also a valid module.
+
+.. code-block:: vyper
+
+    # ownable.vy
+
+    owner: address
+
+    @deploy
+    def __init__():
+        self.owner = msg.sender
+
+    def _check_owner():
+        assert self.owner == msg.sender
+
+    @pure
+    def _times_two(x: uint256) -> uint256:
+        return x * 2
+
+    @external
+    def update_owner(new_owner: address):
+        self._check_owner()
+
+        self.owner = new_owner
+
+This contract basically has two bits of functionality which can be reused upon import, the ``_check_owner()`` function and the ``update_owner()`` function. The ``_check_owner()`` is an internal function which can be used as a helper to check ownership in importing modules, while the ``update_owner()`` is an external function which an importing module can export as an externally facing piece of functionality.
+
+You can use this module's functionality simply by importing it, however any functionality that you do not use from a module will not be included in the final compilation target. For example, if you don't use the ``initializes`` statement to declare a module's location in the storage layout, you cannot use its state. Similarly, if you don't explicitly ``export`` an external function from a module, it will not appear in the runtime code.
+
+Importing a module
+==================
+
+A module can be imported using ``import`` or ``from ... import`` statements. The following are all equivalent ways to import the above module:
+
+.. code-block:: vyper
+
+    import ownable               # accessible as `ownable`
+    import ownable as ow         # accessible as `ow`
+    from . import ownable        # accessible as `ownable`
+    from . import ownable as ow  # accessible as `ow`
+
+When importing using the ``as`` keyword, the module will be referred to by its alias in the rest of the contract.
+
+The ``_times_two()`` helper function in the above module can be immediately used without any further work.
```

Comment:
```suggestion
The ``_times_two()`` helper function in the above module can be immediately used without any further work since it is "pure" and doesn't depend on initialized state.
```